pkg/audit/verifier.go

```diff
@@ -489,6 +499,23 @@ func (verifier *Verifier) GetShare(ctx context.Context, limit *pb.AddressedOrder
 	}, nil
 }
 
+// RemoveFailedPieces removes lost pieces from a pointer
+func (verifier *Verifier) RemoveFailedPieces(ctx context.Context, stripe *Stripe, failedPieces storj.NodeIDList) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	remoteSegment := stripe.Segment.GetRemote()
+	for i, piece := range remoteSegment.GetRemotePieces() {
+		for _, failedPiece := range failedPieces {
+			if piece.NodeId == failedPiece {
+				remoteSegment.RemotePieces = append(remoteSegment.RemotePieces[:i], remoteSegment.RemotePieces[i+1:]...)
+			}
+		}
+	}
+
+	// Update the segment pointer in the metainfo
+	return verifier.metainfo.Put(ctx, stripe.SegmentPath, stripe.Segment)
```

Comment:
I am a little bit worried about how we update the pieces in the pointer. We do the same in the repair too.

What if we the segment is deleted or repaired just before we call `metainfo.Put` here? This would really mess up the pointer and may lead to false negative audits and unfair disqualification of nodes.

Perhaps, we should introduce a `metainfo.Update` method. Let's add a TODO comment about this here, and think about it in another JIRA ticket / PR.

Reply:
https://storjlabs.atlassian.net/browse/V3-2088